streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java

```diff
@@ -356,4 +361,49 @@ public void delete(FlinkCluster cluster) {
     }
     removeById(id);
   }
+
+  /** Check queue label validation when create the cluster if needed. */
+  @VisibleForTesting
+  public void checkQueueValidationIfNeeded(FlinkCluster clusterInfo) {
+    yarnQueueService.checkQueueLabelFormatIfNeeded(
+        clusterInfo.getExecutionModeEnum(), clusterInfo.getYarnQueue());
+    if (!isYarnSessionModeAndNotDefaultQueue(clusterInfo)) {
+      return;
+    }
+    boolean exist = yarnQueueService.existByQueueLabel(clusterInfo.getYarnQueue());
+    ApiAlertException.throwIfFalse(
+        exist, String.format(ERROR_CLUSTER_QUEUE_HINT, clusterInfo.getYarnQueue()));
+  }
+
+  /** Check queue label validation when update the cluster if needed. */
+  @VisibleForTesting
+  public void checkQueueValidationIfNeeded(FlinkCluster oldCluster, FlinkCluster newCluster) {
+    yarnQueueService.checkQueueLabelFormatIfNeeded(
+        newCluster.getExecutionModeEnum(), newCluster.getYarnQueue());
+    if (!isYarnSessionModeAndNotDefaultQueue(newCluster)) {
+      return;
+    }
+
+    if (ExecutionMode.isYarnSessionMode(newCluster.getExecutionModeEnum())
+        && StringUtils.equals(oldCluster.getYarnQueue(), newCluster.getYarnQueue())) {
+      return;
+    }
+    boolean exist = yarnQueueService.existByQueueLabel(newCluster.getYarnQueue());
+    ApiAlertException.throwIfFalse(
+        exist, String.format(ERROR_CLUSTER_QUEUE_HINT, newCluster.getYarnQueue()));
+  }
+
+  /**
+   * Judge the execution mode whether is the Yarn session mode with not default or empty queue
+   * label.
+   *
+   * @param cluster cluster.
+   * @return If the executionMode is yarn session mode and the queue label is not (empty or
+   *     default), return true, false else.
+   */
+  @VisibleForTesting
+  public boolean isYarnSessionModeAndNotDefaultQueue(FlinkCluster cluster) {
```

Comment:
I suggest change `public` to `private`, method rename to `isYarnNotDefaultQueue`